File: 2670/2670.py
class Solution:
    def distinctDifferenceArray(self, nums: List[int]) -> List[int]:
        # One liner
        # return [len(Counter(nums[:i + 1])) - len(Counter(nums[i + 1:])) for i in range(len(nums))]
        prefix = defaultdict(int)
        suffix = Counter(nums)
        result = []
        for x in nums:
            prefix[x] += 1
            suffix[x] -= 1
            if suffix[x] == 0:
                del suffix[x]
            result.append(len(prefix) - len(suffix))
        return result

File: 2670/2670.cpp
class Solution {
public:
    vector<int> distinctDifferenceArray(vector<int>& nums) {
        unordered_map<int, int> prefix, suffix;
        vector<int> result;
        for (int& x: nums) {
            suffix[x]++;
        }
        for (int& x: nums) {
            prefix[x]++;
            if (--suffix[x] == 0) {
                suffix.erase(x);
            }
            result.push_back(prefix.size() - suffix.size());
        }
        return result;
    }
};